### ISSUE TITLE
invalid covariant return type of LTISystem

### ISSUE THREAD

**Issue by yangcyself** [ethz-adrl/control-toolbox#86]
I'm getting `invalid covariant return type` compiler error when builidng a LTISystem:

```cpp
std::shared_ptr<ct::core::ControlledSystem<state_dim, control_dim> > Dynamics(
        new ct::core::LTISystem<state_dim, control_dim>(A, B));
```

```txt
error: invalid covariant return type for ‘const Eigen::Matrix<double, STATE_DIM, STATE_DIM>& ct::core::LTISystem<STATE_DIM, CONTROL_DIM>::getDerivativeState(const ct::core::StateVector<STATE_DIM>&, const ct::core::ControlVector<CONTROL_DIM>&, double)
```

I think the problem comes from the incoherency with StateMatrix StateVector ControlMatrix ControlVector types, and maybe solved as suggested in #39

**Comment by markusgft** [comment]
Dear @yangcyself 
thanks for reporting, I will prioritize this bug in the next release! 
Best

**Comment by markusgft** [comment]
Will be fixed in v3.0.3, see also commit 5f73b9e on the devel branch.
